docs/developer-docs/backend/arguments.mdx

```diff
@@ -0,0 +1,262 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Passing arguments
+
+## Overview 
+
+This document aims to provide examples of passing different argument types to a canister.
+
+## Arrays
+
+An `array` is a collection of items of the same data type.
+
+The following example defines an array of numbers and a function that returns the array.
+
+<Tabs groupId="language">
+<TabItem value="motoko" label="Motoko"t>
+
+Motoko differentiates between immutable arrays, which cannot be altered, and mutable arrays, which can be modified.
+
+To declare an immutable array (default):
+
+```motoko
+let a : [Nat] = [0, 1, 2, 3];
+```
+
+To declare a mutable array, the [var _] syntax is used:
+
+```
+let a : [var Nat] = [var 0, 1, 2, 3] ;
+```
+
+To declare a function that returns an array:
+
+```
+public func get_numbers(a: [Nat]) : [Nat] {
+    return a;
+}
+```
+
+The `Array` Motoko base library provides utility functions on arrays. To learn more about the `Array` Motoko base library, refer to the [Motoko base library reference on Array](https://internetcomputer.org/docs/current/motoko/main/base/Array) and the [Motoko language quick reference on arrays](https://internetcomputer.org/docs/current/motoko/main/language-manual/#arrays).
+
+</TabItem>
+
+<TabItem value="rust" label="Rust">
+
+Rust uses the `vec` type to represent vectors (sequences, lists, arrays)
+
+```rust
+let numbers = vec![0, 1, 2, 3 ];
+#[query]
+fn get_numbers(numbers: Vec<u32>) -> Vec<u32> {
+    numbers
+}
+```
+
+</TabItem>
+
+<TabItem value="typescript" label="Typescript">
+
+Azle refers to the `Vec` type to represent the equivalent of an `array` in Typescript. This is because `Vec` aligns with the Candid type.
```

Comment:
```suggestion
Azle refers to the `Vec` type to represent the equivalent of an `array` in Typescript. This is because `Vec` aligns with the [Candid](/docs/current/developer-docs/backend/candid/candid-concepts) type.
```